daml-script/dump/src/main/scala/com/daml/script/dump/Encode.scala

```diff
@@ -18,13 +19,34 @@ import scalaz.std.set._
 import scalaz.syntax.foldable._
 
 private[dump] object Encode {
-  def encodeTransactionTreeStream(trees: Seq[TransactionTree]): Doc = {
-    val parties = trees.toList.foldMap(partiesInTree(_))
+  def encodeTransactionTreeStream(
+      acs: Map[String, CreatedEvent],
+      trees: Seq[TransactionTree],
+  ): Doc = {
+    val parties = partiesInContracts(acs.values) ++ trees.toList.foldMap(partiesInTree(_))
     val partyMap = partyMapping(parties)
-    val cids = trees.map(treeCreatedCids(_))
-    val cidRefs = trees.toList.foldMap(treeReferencedCids(_))
-    val cidMap = cidMapping(cids, cidRefs)
-    val refs = trees.toList.foldMap(treeRefs(_))
+
+    val acsCidRefs = acs.values.toList.foldMap(createdReferencedCids)
+    val treeCidRefs = trees.toList.foldMap(treeReferencedCids)
+    val cidRefs = acsCidRefs ++ treeCidRefs
+
+    val archivedCidRefs = acsCidRefs -- acs.keySet
```

Comment:
Given that these are not just archived contracts but also divulged contracts or simply unknown contracts (you can pass in an arbitrary string via the ledger API), maybe call this `unknownCidRifs` or `missingCidRefs`.